dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/cache/AnalyticsCache.java

```diff
@@ -64,17 +70,27 @@ public AnalyticsCache( final CacheProvider cacheProvider,
         checkNotNull( cacheProvider );
         checkNotNull( analyticsCacheSettings );
 
+        this.cacheProvider = cacheProvider;
         this.analyticsCacheSettings = analyticsCacheSettings;
-        long initialExpirationTime = analyticsCacheSettings.fixedExpirationTimeOrDefault();
-        this.queryCache = cacheProvider.createAnalyticsResponseCache(
-            Duration.ofSeconds( initialExpirationTime ) );
+    }
+
+    private synchronized void initializeCache()
+    {
+        if ( queryCache == null )
+        {
+            long initialExpirationTime = analyticsCacheSettings.fixedExpirationTimeOrDefault();
+            this.queryCache = cacheProvider.createAnalyticsResponseCache(
+                Duration.ofSeconds( initialExpirationTime ) );
 
-        log.info( String.format( "Analytics server-side cache is enabled with expiration time: %d s",
-            initialExpirationTime ) );
+            log.info( String.format( "Analytics server-side cache is enabled with expiration time: %d s",
+                initialExpirationTime ) );
+        }
     }
 
     public Optional<Grid> get( final String key )
     {
+        initializeCache();
```

Comment:
Do we need to call `initializeCache()` before every method call?
Could we call `initializeCache()` only once inside the `constructor`, and remove all others calls inside each method?

Reply:
Unfortunately not, this is the reason for the refactor since we can't do database queries in bean constructors.